gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/sideEffect/Aggregate.feature

```diff
@@ -144,145 +144,490 @@ Feature: Step - aggregate()
+    | v[lop] |
+    | null |
+    | null |
+    | null |
 
   Scenario: g_V_aggregateXaX_hasXperson_age_gteX30XXX_capXaX_unfold_valuesXnameX
     Given the modern graph
     And the traversal of
+    """
+    g.V().aggregate("a").has("person","age", P.gte(30)).cap("a").unfold().values("name")
+    """
+    When iterated to list
+    Then the result should be unordered
+    | result |
+    | marko |
+    | josh |
+    | peter |
+    | lop |
+    | vadas |
+    | ripple |
+
+  Scenario: g_withSideEffectXa_1_sumX_V_aggregateXaX_byXageX_capXaX
+    Given the modern graph
+    And the traversal of
+    """
+    g.withSideEffect("a", 1, Operator.sum).V().aggregate("a").by("age").cap("a")
+    """
+    When iterated to list
+    Then the result should be unordered
+    | result   |
+    | d[124].i |
+
+  Scenario: g_withSideEffectXa_1_sumX_V_aggregateXlocal_aX_byXageX_capXaX
+    Given the modern graph
+    And the traversal of
+    """
+    g.withSideEffect("a", 1, Operator.sum).V().aggregate(Scope.local, "a").by("age").cap("a")
+    """
+    When iterated to list
+    Then the result should be unordered
+      | result   |
+      | d[124].i |
+
+
+  Scenario: g_withSideEffectXa_1_sumLongX_V_aggregateXaX_byXageX_capXaX
+    Given the empty graph
+    And the graph initializer of
+    # This graph is the same as ModernGraph, but ages are stored as Long.
+      """
+      g.addV("person").property("name", "marko").property("age", 29L).as("marko").
+        addV("person").property("name", "vadas").property("age", 27L).as("vadas").
+        addV("software").property("name", "lop").property("lang", "java").as("lop").
+        addV("person").property("name","josh").property("age", 32L).as("josh").
+        addV("software").property("name", "ripple").property("lang", "java").as("ripple").
+        addV("person").property("name", "peter").property("age", 35L).as('peter').
+        addE("knows").from("marko").to("vadas").property("weight", 0.5d).
+        addE("knows").from("marko").to("josh").property("weight", 1.0d).
+        addE("created").from("marko").to("lop").property("weight", 0.4d).
+        addE("created").from("josh").to("ripple").property("weight", 1.0d).
+        addE("created").from("josh").to("lop").property("weight", 0.4d).
+        addE("created").from("peter").to("lop").property("weight", 0.2d)
       """
-      g.V().aggregate("a").has("person","age", P.gte(30)).cap("a").unfold().values("name")
+    And the traversal of
+    """
+    g.withSideEffect("a", 1L, Operator.sumLong).V().aggregate("a").by("age").cap("a")
+    """
+    When iterated to list
+    Then the result should be unordered
+      | result   |
+      | d[124].l |
+
+  Scenario: g_withSideEffectXa_1_sumLongX_V_aggregateXlocal_aX_byXageX_capXaX
+    Given the empty graph
+    And the graph initializer of
+    # This graph is the same as ModernGraph, but ages are stored as Long.
+      """
+      g.addV("person").property("name", "marko").property("age", 29L).as("marko").
```

Comment:
OK I  think it doesn't matter because when using `sumLong` operator, values are casted to `long` anyway. I will just use the regular Modern graph and see if the test passes.